File: formation/intro_to_linked_lists.js


class Node {
    constructor(val, next=null){
        this.val = val
        this.next = next 
    }
}

let head = 
    new Node(5,
        new Node(10,
            new Node(15,
                new Node(20,
                    new Node(25,)))))

let cur = head
while(cur !== null){
    console.log(cur.val)
    cur = cur.next 
}

console.log("All Done")



// Alternative - This will stop the loop one before the end
// So that you can capture the last value and do something with it
// let cur = head
// while(cur.next !== null){
//     console.log(cur.val)
//     cur = cur.next
// }

// if(cur !== null){
//     console.log("Not done yet:", cur.val)
// }

File: formation/intro_to_linked_lists.py


class Node:
    def __init__(self, val, next=None):
        self.val = val
        self.next = next 

head = Node(5,
            Node(10,
                Node(15,
                    Node(20,
                        Node(25)))))

cur = head
while cur:
    print(cur.val)
    cur = cur.next

print("All Done")



# Alternative - This will stop the loop one before the end
# So that you can capture the last value and do something with it
cur = head
while cur.next:
    print(cur.next)
    cur = cur.next 
if cur != None:
    print("Not done yet:", cur.val)